test/ntttcp_test.py

```diff
@@ -0,0 +1,470 @@
+        throughput_in_jsonfile = throughput_in_jsonfile_pattern.findall(value_filename_out)
+        if subprocess.run(f"ls {filename}", shell=True, check=True) and json.loads(value_filename_out, strict=False):
+            if throughput_in_jsonfile:
+                print("throughput_in_jsonfile:", throughput_in_jsonfile)
+                return True
+            else:
+                return False
+        else:
+            return False
+
+    def get_starting_port_number(self) -> int:
+        # DBG : New connection: local:48536 [socket:4] --> 127.0.0.1:10000
+        starting_port_pattern = re.compile(r"DBG : New connection: local:(\d+) \[socket:4\] --> 127.0.0.1:(\d+)")
+        starting_port = starting_port_pattern.findall(self.send_stdout)
+        if starting_port:
+            print("starting_port:", starting_port[0][1])
+            return int(starting_port[0][1])
+        else:
+            return False
+
+    def get_ports_numbers(self) -> int:
+        # 800 connections tested
+        mapping_option_value_pattern = re.compile(r"(\d+) connections tested")
+        mapping_option_value = mapping_option_value_pattern.findall(self.send_stdout)
+        if mapping_option_value[0]:
+            return int(mapping_option_value[0])
+        else:
+            return False
+
+    def is_having_system_tcp_retransmit(self) -> bool:
+        # INFO:   retrans_segments/sec   :2018.11
+        system_tcp_retransmit_pattern = re.compile(r"retrans_segments/sec\s+:(\d+\.\d+)")
+        system_tcp_retransmit = system_tcp_retransmit_pattern.findall(self.send_stdout)
+        if system_tcp_retransmit:
+            print("system_tcp_retransmit:", system_tcp_retransmit[0])
+            return True
+        else:
+            return False
+
+    def is_show_nic_packets(self) -> bool:
+        # INFO:   tx_packets     :329598
+        tx_packets_pattern = re.compile(r"\s+tx_packets\s:(\d+)")
+        tx_packets = tx_packets_pattern.findall(self.send_stdout)
+        # INFO:   rx_packets     :329598
+        rx_packets_pattern = re.compile(r"\s+rx_packets\s:(\d+)")
+        rx_packets = rx_packets_pattern.findall(self.send_stdout)
+        if tx_packets and rx_packets and int(tx_packets[0]) > 100000 and int(rx_packets[0]) > 100000:
+            print("tx_packets:", tx_packets[0])
+            print("rx_packets:", rx_packets[0])
+            return True
+        else:
+            return False
+
+    def is_show_dev_interrupts(self) -> bool:
+        # INFO:   total          :15
+        interrupts_pattern = re.compile(r"total\s+:(\d+)")
+        interrupts = interrupts_pattern.findall(self.send_stdout)
+        if interrupts and int(interrupts[0]) > 0:
+            print("interrupts:", interrupts[0])
+            return True
+        else:
+            return False
+
+class TestNtttcp:
+    expected_throughput = 70  # need evaluate
```

Comment:
Can you please evaluate and confirm this baseline value (70Gbps?).

Reply:
done